core/libp2p/network/impl/connection_manager_impl.cpp

```diff
@@ -94,7 +94,10 @@ namespace libp2p::network {
   ConnectionManagerImpl::ConnectionManagerImpl(
       std::shared_ptr<peer::AddressRepository> addr,
       std::shared_ptr<TransportManager> tmgr)
-      : addr_repo_(std::move(addr)), transport_manager_(std::move(tmgr)) {}
+      : addr_repo_(std::move(addr)), transport_manager_(std::move(tmgr)) {
+    BOOST_ASSERT(addr_repo_ != nullptr);
```

Comment:
```suggestion
    BOOST_ASSERT(addr_repo_);
```
Is enough

Reply:
it is enough, but it is better to explicitly compare. btw clang-tidy would complain that ptr is casted to bool